src/java/htsjdk/samtools/cram/encoding/reader/CramRecordReader.java

```diff
@@ -190,6 +197,7 @@ public void read(final CramCompressionRecord cramRecord) {
             if (prevRecord != null)
                 System.err.printf("Failed at record %d. Here is the previously read record: %s\n", recordCounter,
                         prevRecord.toString());
+            if (e instanceof SAMFormatException) throw (SAMFormatException)e;
```

Comment:
Add a separate catch block for `SAMFormatException` instead of calling `instanceof` on the weakly-typed exception object.